### DISCUSSION TITLE
Why was IDEA moved to legacy provider？

### DISCUSSION THREAD

**Discussion by wujiening** [openssl/openssl#24683] [Q&A] [answered]
My native language is not English. If there are any errors in the description of this issue, I apologize in advance.
I need to use IDEA in my work, but I found that IDEA has been moved to the legacy provider in OpenSSL 3.
The keys in the legacy provider are all discarded by OpenSSL. I understand that abandoned algorithms have security issues (such as DES).
My question:
Why is the IDEA algorithm being abandoned? Is IDEA also abandoned due to security issues? What security issues does IDEA have?
Thanks.

**Comment by levitte** [comment]
> The keys in the legacy provider are all discarded by OpenSSL.

This part, I don't understand.  Can you show an example that demonstrates this?

**Reply by wujiening** [reply]
There may be something wrong with my expression here. I mean, the algorithm in legacy provider module is not recommended, and it may be abandoned in future updates of openssl.

In my understanding, the algorithm in the deault provider module is recommended, but the algorithm in the legacy provider is not recommended.

**Reply by t8m** [reply, marked as answer]
IDEA has block size of 64 bits which on its own makes it a weak cipher by current standards.

**Reply by t8m** [reply]
Of course 3DES has also just 64 bit block size which means it should eventually be moved to the legacy provider as well. Unfortunately it was still used too widely when 3.0 was released.

**Reply by levitte** [reply]
> There may be something wrong with my expression here. I mean, the algorithm in legacy provider module is not recommended, and it may be abandoned in future updates of openssl.

Ah, you referred to the future.  I read it as something that's happening to you now when trying to use it.
Thanks for the clarification!
